### PR TITLE
cryptonote: add function to get weight from a pruned tx

### PR DESCRIPTION
**Pull request by moneromooo-monero**
The weight of the prunable data is deterministic from the
unpruned data, so it can be determined from a pruned tx

### CODE REVIEW

**Comment by stoffu** [comment]
The build is failing because this PR references CSLAG which isn't yet merged.

File: src/cryptonote_basic/cryptonote_format_utils.cpp [hunk 1]
```diff
@@ -386,27 +406,64 @@ namespace cryptonote
...
+
+    // nbps (technically varint)
+    weight += 1;
+
+    // calculate deterministic bulletproofs size (assumes canonical BP format)
+    size_t nrl = 0, n_padded_outputs;
+    while ((n_padded_outputs = (1u << nrl)) < tx.vout.size())
+      ++nrl;
+    nrl += 6;
+    extra = 32 * (9 + 2 * nrl) + 2;
```
[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply src/cryptonote_basic/cryptonote_format_utils.cpp]: They are the array lengths for L and R. Technically varints, but the array are log sized so they'll never to above one byte each.

[the remark this answers is not in the document]
Comment (stoffu) [reply src/cryptonote_basic/cryptonote_format_utils.cpp]: Ah, thanks.

File: src/cryptonote_basic/cryptonote_format_utils.cpp [hunk 2]
```diff
@@ -386,27 +406,61 @@ namespace cryptonote
...
     CHECK_AND_ASSERT_THROW_MES_L1(bp_clawback <= std::numeric_limits<uint64_t>::max() - blob_size, "Weight overflow");
     return blob_size + bp_clawback;
   }
   //---------------------------------------------------------------
+  uint64_t get_pruned_transaction_weight(const transaction &tx)
+  {
+    CHECK_AND_ASSERT_MES(tx.pruned, std::numeric_limits<uint64_t>::max(), "get_pruned_transaction_weight does not support non pruned txes");
+    CHECK_AND_ASSERT_MES(tx.version >= 2, std::numeric_limits<uint64_t>::max(), "get_pruned_transaction_weight does not support v1 txes");
+    CHECK_AND_ASSERT_MES(tx.rct_signatures.type >= rct::RCTTypeBulletproof2,
+        std::numeric_limits<uint64_t>::max(), "get_pruned_transaction_weight does not support older range proof types");
```
Comment (stoffu) [inline src/cryptonote_basic/cryptonote_format_utils.cpp]: Why this restriction? Prunable tx weights are always deterministic and should be computable for older range sig types.

Comment (moneromooo-monero) [reply src/cryptonote_basic/cryptonote_format_utils.cpp]: To save time, since it's only used for recent txes.

**Pull request merged**